### PR TITLE
Enforce good practices v2

### PR DESCRIPTION
**Pull request by smeriwether**
This PR makes 2 changes (1 a lot larger than the other)

1. Require success! or fail! when creating a command with #new

A previous commit said that all commands now require a call to
`success!` or `fail!` and would fail if not. Unfortunately, that was
only true when using a command in the form `CommandName.call` but not
`CommandName.new.call`. See commit 137ffb43f.

2. successful? alias for success?

When writing tests against Chief::Commands I frequently write
`expect(result).to be_success`
but I would rather write
`expect(result).to be_successful`
See commit e5a0c37b1da

### CODE REVIEW

File: lib/chief/command.rb [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
-    def self.value(*args)
-      call(*args).value
+    def self.value(*args, &block)
+      call(*args, &block).value
     end
 
-    def call
-      fail NotImplementedError, 'Implement #call in subclass'
-    end
+    def self.inherited(subclass)
```
Comment (kellywu) [inline lib/chief/command.rb:11]: this is cool, i didn't know about this

File: lib/chief/command.rb [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
-    end
+        def success!(value = true)
+          create_result_object(value, nil)
+        end
 
-    def fail!(value = false, errors = true)
-      create_result_object(value, errors)
-    end
+        def fail!(value = false, errors = true)
+          create_result_object(value, errors)
```
Comment (kellywu) [inline lib/chief/command.rb:28]: nothing to related to this PR, but just something to think about as u see more usages of chief.

i think the way we use failure objects is a bit confusing and can be simplified/improved

File: lib/chief/command.rb [hunk 3]
```diff
@@ -1,38 +1,42 @@
...
+        end
 
-    private
+        private
 
-    attr_reader :created_result_object
+        def create_result_object(value, errors)
+          @fullfilled = true
```
Comment (kellywu) [inline lib/chief/command.rb:34]: super nitpick, but you have an accessor, do we want to use the setter here? (e.g. no `@`)

File: spec/chief/integration_spec.rb [hunk 4]
```diff
@@ -25,67 +25,122 @@ def call
...
+  end
 
-        expect(result).to be_a Result
-        expect(result).to be_success
-        expect(value).to eq :some_value
-        expect(errors).to be_nil
-      end
+  describe "Commands that are instantiated with new and don't return a Chief::Result object" do
+    it 'should raise an error' do
+      expect { Example::CommandWithoutChiefResult.new.call }.to raise_error(RuntimeError)
```
Comment (kellywu) [inline spec/chief/integration_spec.rb]: 👏

**Review by kellywu — APPROVED** [review]
Great improvements to enforce usage

File: lib/chief/command.rb (hunk 2 above)
Comment (stephenmhoc) [reply lib/chief/command.rb:28]: I completely agree

**Pull request merged**